app/api/puzzle/submit/route.ts

```diff
@@ -45,6 +58,16 @@ export async function POST(req: NextRequest) {
     );
   }
 
+  // Delete active puzzle, as it is solved.
+  const activePuzzle = await ActivePuzzleColl.findOneAndDelete({ username: user.username });
```

Comment:
Doing this in `submit`, which is invoked when the user gets the puzzle wrong, means that users who just got a puzzle wrong but still are working on it will get a different one on refresh. This, however, is the behaviour of Lichess, so I think it is okay.